protected.go

```diff
@@ -53,14 +53,21 @@ func (addr *protectedAddr) TCPAddr() *net.TCPAddr {
 }
 
 // New construct a protector from the protect function and DNS server IP address.
-func New(protect Protect, dnsServerIP string) *Protector {
-	ipAddr := net.ParseIP(dnsServerIP)
+func New(protect Protect, dnsServer string) *Protector {
+	var ipAddr net.IP
+	host, port, err := splitHostPort(dnsServer)
+	if err != nil {
+		log.Errorf("Invalid DNS server address %s: %v", dnsServer, err)
+	} else {
+		ipAddr = net.ParseIP(host)
+	}
 	if ipAddr == nil {
-		log.Debugf("Invalid DNS server IP %s, default to %s", dnsServerIP, defaultDNSServer)
+		log.Debugf("Using default DNS server: %v", defaultDNSServer)
 		ipAddr = net.ParseIP(defaultDNSServer)
+		port = dnsPort
 	}
-
-	dnsAddr := syscall.SockaddrInet4{Port: dnsPort}
+	log.Debugf("Using dns server %v:%v", ipAddr, port)
+	dnsAddr := syscall.SockaddrInet4{Port: port}
```

Comment:
The logging should make this clear, but here's the main change to use the correct port